src/middleware/secure-headers/secure-headers.ts

```diff
@@ -255,6 +272,36 @@ function getCSPDirectives(
       ]
 }
 
+function getPermissionsPolicyDirectives(policy: PermissionsPolicyOptions): string {
+  return Object.entries(policy)
+    .map(([directive, value]) => {
+      if (typeof value === 'boolean') {
+        return `${camelToKebab(directive)}=${value ? '()' : 'none'}`
+      }
+      if (Array.isArray(value)) {
+        if (value.length === 0) {
+          return `${camelToKebab(directive)}=()`
+        }
+        const allowlist = value.map((v) => (v === 'self' || v === 'src' ? v : v)).join(' ')
+        return `${camelToKebab(directive)}=(${allowlist})`
+      }
+      return ''
+    })
+    .filter(Boolean)
+    .join(', ')
+}
+
+function camelToKebab(str: string): string {
```

Comment:
By converting uppercase letters that follow lowercase letters or numbers, the implementation becomes simpler
```ts
function camelToKebab(str: string): string {
  return str.replace(/([a-z\d])([A-Z])/g, '$1-$2').toLowerCase()
}
```